pkg/agent/proxy/instance.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/record"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
+	"github.com/vmware-tanzu/antrea/pkg/agent/proxy/upstream/config"
+	"github.com/vmware-tanzu/antrea/pkg/agent/querier"
+)
+
+const (
+	resyncPeriod  = time.Minute
+	componentName = "antrea-agent-proxy"
+)
+
+type Instance struct {
```

Comment:
I actually didn't mean the Interface "Instance" only but also the struct "instance", it seems to me it does nothing but does some initialization which could also be part of proxier.
There is no difference if moving `Run` to proxier and merging `New` with `newProxier`.
I think the upstream has an outer struct for proxy.Provider is because it has multiple implementation so it can put shared part outside, which doesn't apply to antrea.

Reply:
Got it, updated.